python/ray/worker.py

```diff
@@ -1523,6 +1523,7 @@ def process_task(task): # wrapping these lines in a function should cause the lo
                                         "function_name": function_name})
 
   check_main_thread()
+  worker.photon_client.register_pid()
```

Comment:
Oh yeah, makes sense to do it as part of `photon_connect`.

The local scheduler won't ever terminate the driver. It only terminates child processes. If you look at the handler for `REGISTER_PID` messages, it matches up worker clients with child processes according to pid.